rfcs/20200929-keras-mixed-precision.md

```diff
@@ -0,0 +1,1178 @@
+
+Throughout this proposal, the phrase "float16" will sometimes be used to refer to either float16 or bfloat16 in order to avoid repeating the phrase "float16 and bfloat16" throughout the doc.
+
+## Dtype policies overview
+
+Every layer will have a special object called a "dtype policy", determining the dtype of the layer’s computations and variables. A dtype policy additionally holds a LossScale object, which is described in the [Loss Scaling section](#design-proposal-loss-scaling) of this RFC. The dtype policy class will be exposed as `tf.keras.mixed_precision.Policy`, and is currently exposed in TensorFlow 2.3 as [`tf.keras.mixed_precision.experimental.Policy`](https://www.tensorflow.org/api_docs/python/tf/keras/mixed_precision/experimental/Policy).
+
+Each policy has a `compute_dtype` field and a `variable_dtype` field. The compute dtype specifies the dtype of a layer’s operations. A layer’s output is typically in the compute dtype. The variable dtype is the default dtype of the layer’s variables, although in a few rare cases a layer may create variables of a different dtype.
+
+Despite the word "mixed_precision" in the API name, all layers, even non-mixed precision layers, have a dtype policy. However, the only motivation for layers having a dtype policy instead of a single dtype is to support mixed precision, and only mixed precision users need to directly interact with policies, which is why the word "mixed_precision" appears in the API name.
+
+The constructor for a policy is:
+
+```python
+class Policy:
+
+  def __init__(self, name, loss_scale='auto'):
+    ...
+```
+
+The `name` argument determines the compute dtype and the variable dtype. It also determines the loss scale of the policy if the user does not pass in a loss scale to the constructor. The possible values for `name` are:
+
+* Any dtype: The compute and variable dtype is `name`. By default, there is no loss scale. `name` can be a DType, or any value convertible to a DType (including strings).
+* `"mixed_float16"`: The compute dtype is float16. The variable dtype is float32. The default loss scale is "dynamic".
+* `"mixed_bfloat16"`: The compute dtype is bfloat16. The variable dtype is float32. There is no default loss scale, as loss scaling is only useful when float16 is used.
+
+Unlike most TensorFlow functions with a `name` argument, the Policy `name` argument has a semantic impact on the TensorFlow program, and is not just used to uniquely identify an op or layer. The word "name" is chosen for lack of a better word to call the argument.
```

Comment:
The issue is the argument is also exposed as as a property, so if it were called `dtype` instead of `name`, Policy would have a `dtype` field, a `compute_dtype` field, and a `variable_dtype` field. It would be nonobvious that the `dtype` field was mostly unused and only used to determine the `compute_dtype` and `variable_dtype` fields.

Reply:
Thanks for the explanation, that makes sense :+1: